core/src/expand.rs

```diff
@@ -149,9 +155,16 @@ pub(crate) fn expand(
     })
 }
 
+/// This function takes the list of query fragments and substitutes all `{#binding_name}`
+/// occurrences with their literal strings from the respective match statements.
+///
+/// These literal strings however, can once again contain another `{#binding_name}`, which is why
+/// this function is called from a while loop.
 fn expand_compile_time_bindings(
+    // REVIEW: IIUC I would rename `unexpanded_fragments` to `fragments`, as they can theoretically
+    // REVIEW: contain fragments that're already fully expanded. The name suggests otherwise.
     unexpanded_fragments: Vec<syn::LitStr>,
```

Comment:
They're unexpanded as far as this function is concerned.  It has no knowledge of previous expansion passes, it just knows that these are fragments that need to be expanded.

It was originally called just `fragments` but I found the function body a bit more confusing then.  Now it's just looping over it in one place though so doesn't really matter and I think it's fine to rename it back.